app/components/content_list_item_component.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class ContentListItemComponent < ViewComponent::Base
+  def initialize(content_list_item:, viewer_class:)
+    @file = content_list_item
+    @viewer_class = viewer_class
+  end
+  attr_reader :file, :viewer_class
+
+  def url
+    file.file_url
+  end
+
+  def download_label
+    file.label_or_filename
+  end
+
+  # Note: when other content types make use of this component, add other JS methods and set them below
+  # the JS method to call in content_list_controller.js which will be responsible for swapping out the viewer with the selected file
+  def show_method
+    case viewer_class
+      when "Embed::Viewer::PdfViewer"
+        return 'showPdf'
+     end
+  end
+
+  # Only PDF content is rendered in this panel for now
+  # Media content also does not need this component since it uses custom javascript to render thumbnails and link to content
+  def render?
+    return true if viewer_class == "Embed::Viewer::PdfViewer" and url.ends_with?('.pdf') # do not render any file that is not a PDF
+
+    false
+  end
+end
```

Comment:
New components which render the list of Contents in the side panel.  Designed to be flexible when extending to new content types